tests/unit/experiments/providers/simple_experiment_tests.py

```diff
@@ -0,0 +1,331 @@
+                     "test takes too long to run for normal local iteration")
+    def test_calculate_bucket_with_seed(self):
+        seeded_cfg = {
+            "id": 1,
+            "name": "test_experiment",
+            "owner": "test",
+            "type": "single_variant",
+            "version": "1",
+            "start_ts": time.time() - THIRTY_DAYS,
+            "stop_ts": time.time() + THIRTY_DAYS,
+            "enabled": True,
+            "experiment": {
+                "variants": [
+                    {
+                        "name":"variant_1",
+                        "size":0.1,
+                    },
+                    {
+                        "name":"variant_2",
+                        "size":0.1,
+                    },
+                ],
+                "experiment_version":1,
+                "shuffle_version":1,
+                "bucket_seed": "some_new_seed",
+            }
+        }
+
+        experiment = parse_experiment(seeded_cfg)
+
+        # Give ourselves enough users that we can get some reasonable amount of
+        # precision when checking amounts per bucket.
+        num_users = experiment.num_buckets * 2000
+        fullnames = []
+        for i in range(num_users):
+            fullnames.append("t2_%s" % str(i))
+
+        counter = collections.Counter()
+        bucketing_changed = False
+        for fullname in fullnames:
+            self.assertEqual(experiment.seed, "some_new_seed")
+            bucket1 = experiment._calculate_bucket(fullname)
+            counter[bucket1] += 1
+            # Ensure bucketing is deterministic.
+            self.assertEqual(bucket1, experiment._calculate_bucket(fullname))
+
+            current_seed = experiment.seed
+            experiment._seed = "newstring"
+            bucket2 = experiment._calculate_bucket(fullname)
+            experiment._seed = current_seed
+            # check that the bucketing changed at some point. Can't compare
+            # bucket1 to bucket2 inline because sometimes the user will fall
+            # into both buckets, and test will fail. 
+            if bucket1 != bucket2:
+                bucketing_changed = True
+
+        self.assertTrue(bucketing_changed)
+
+        for bucket in range(experiment.num_buckets):
+            # We want an even distribution across buckets.
+            expected = num_users / experiment.num_buckets
+            actual = counter[bucket]
+            # Calculating the percentage difference instead of looking at the
+            # raw difference scales better as we change NUM_USERS.
+            percent_equal = float(actual) / expected
+            self.assertAlmostEqual(percent_equal, 1.0, delta=.10,
+                                   msg='bucket: %s' % bucket)
+
+    def test_variant_returns_none_if_out_of_time_window(self):
+        valid_cfg = get_simple_config()
+        experiment_valid = parse_experiment(valid_cfg)
+        experiment_valid._choose_variant = choose_variants_override
```

Comment:
Rather than monkey patching the function like this, you should decorate the test with a `mock.patch` decorator like and set the `return_value` on the mock, i.e.:

```python
@mock.patch('baseplate.experiments.providers.simple_experiment._choose_variant')
def test_variant_returns_none_if_out_of_time_window(self, choose_variant_mock):
    choose_variant_mock.return_value = 'fake_variant'
```

Reply:
added